src/casp/cell_data_manager/README.md

```diff
@@ -0,0 +1,118 @@
+# Cell Data Manager Module
+
+This module is designed for managing cell data. Currently, we store cell data in BigQuery, which uses its own SQL dialect to handle data in the warehouse.
+This module should not be limited to only BigQuery, as there is potential to switch to other data warehouse products.
+
+For more information about BigQuery SQL dialect, please visit [this link](https://cloud.google.com/bigquery/docs/introduction-sql).
+
+## SQL Query Management
+
+Template generator is used for managing SQL queries due to the following reasons:
+
+1. `cellarium-cloud` relies on a large number of queries with dynamic content (e.g., optional joins, different filters in `where` clauses).
+2. To maintain the DRY (Don't Repeat Yourself) principle in SQL queries.
+3. To separate SQL code from Python files, which makes our codebase cleaner, more modular, and thus more easily maintainable.
+
+### Template Generator
+
+[Mako template generator](https://www.makotemplates.org/) is used for this purpose. I considered both [Mako](https://www.makotemplates.org/) and [Jinja2](https://jinja.palletsprojects.com/en/2.10.x/), which were originally designed for HTML file templating. However, they seem suitable for managing SQL queries as well. You can learn more from these references:
+
+- [Medium: Jinja + SQL = <3](https://medium.com/p/7e4dff8d8778)
+- [Medium: Jinja the SQL way of Ninja](https://medium.com/analytics-and-data/jinja-the-sql-way-of-the-ninja-9a64fc815564)
+- [Pushmetrics: Why Jinja and SQL?](https://pushmetrics.io/learn/jinja/why-jinja-and-sql/)
+
+There are also wrappers around both Jinja and Mako that are specifically used for SQL management:
+
+- [Quma (Mako wrapper)](https://github.com/ebenefuenf/quma)
+- [JinjaSQL (Jinja wrapper)](https://github.com/sripathikrishnan/jinjasql)
+
+However, it appears that these libraries are not well-maintained and/or have few stars on GitHub.
+
+### Why Mako?
+
+In my personal opinion, [Mako template generator](https://www.makotemplates.org/) seems to have a better syntax. It is easier to use and has a more Pythonic approach. It also allows for the seamless integration of Python helper functions within templates.
```

Comment:
In our opinion :)

This is not a blog post Fedor :p

Reply:
Lol, gotcha.:)